Remotion/Validation/UnitTests/RoleCollectors/AddingPropertyValidationRuleCollector1Test.cs

```diff
@@ -105,51 +124,155 @@ public void Create_PropertyDeclaredInBaseClass ()
     {
+      var validationMessageMock = new Mock<IValidationMessageFactory>();
+      validationMessageMock
+          .Setup(_ => _.CreateValidationMessageForPropertyValidator(It.IsAny<NotNullValidator>(), _property))
+          .Returns((ValidationMessage)null);
+
+      Func<PropertyValidationRuleInitializationParameters, IPropertyValidator> validatorFactory = param => new NotNullValidator(param.ValidationMessage);
+
+      _addingPropertyValidationRuleCollector.RegisterValidator(validatorFactory);
+
+      Assert.That(
+          () => _addingPropertyValidationRuleCollector.CreateValidationRule(validationMessageMock.Object),
+          Throws.InvalidOperationException
+              .With.Message.EqualTo("The IValidationMessageFactory did not return a result for NotNullValidator applied to property 'UserName' on "
+                                    + "type 'Remotion.Validation.UnitTests.TestDomain.Customer'."));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CreateValidationRule_WithConditionNotNull_InitializesConditionForCreatedPropertyValidationRule ()
     {
+      var validationMessage = new DelegateBasedValidationMessage(() => "Test");
+
+      var validationMessageMock = new Mock<IValidationMessageFactory>();
+      validationMessageMock
+          .Setup(e => e.CreateValidationMessageForPropertyValidator(It.IsAny<NotNullValidator>(), _property))
+          .Returns(validationMessage);
+
+      Func<PropertyValidationRuleInitializationParameters, IPropertyValidator> validatorFactory = param => new NotNullValidator(param.ValidationMessage);
+      Func<Customer, bool> predicate = _ => true;
+
+      _addingPropertyValidationRuleCollector.RegisterValidator(validatorFactory);
+      _addingPropertyValidationRuleCollector.SetCondition(predicate);
+
+      var result = _addingPropertyValidationRuleCollector.CreateValidationRule(validationMessageMock.Object);
+
+      Assert.That(((PropertyValidationRule<Customer, string>)result).Condition, Is.SameAs(predicate));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void SetCondition ()
     {
+      Func<Customer, bool> predicate = _ => true;
+      _addingPropertyValidationRuleCollectorType.SetCondition(predicate);
+
+      Assert.That(_addingPropertyValidationRuleCollectorType.Condition, Is.SameAs(predicate));
+    }
+
+    [Test]
+    public void SetCondition_WithInvalidTypeOfPredicate_ThrowsArgumentException ()
+    {
+      Func<Person, bool> predicate = _ => true;
+
+      Assert.That(
+          () => _addingPropertyValidationRuleCollector.SetCondition(predicate),
+          Throws.ArgumentException
+              .With.Message.EqualTo("The type 'Remotion.Validation.UnitTests.TestDomain.Person' of the predicate does not match the "
+                                    + "type 'Remotion.Validation.UnitTests.TestDomain.Customer' of the validation rule."));
```

Comment:
move message to new line. ALso, please use the argumentmessage-check API and also assert the parameter name for completeness' sake.